core/lib/network-recorder.js

```diff
@@ -243,9 +244,8 @@ class NetworkRecorder extends RequestEventEmitter {
     if (record.redirectSource) {
       return record.redirectSource;
     }
-    const stackFrames = record.initiator.stack?.callFrames || [];
-    const initiatorURL = record.initiator.url || stackFrames[0]?.url;
 
+    const initiatorURL = PageDependencyGraph.getNetworkInitiators(record)[0];
```

Comment:
I was originally going to manually check `record.initiator.stack.parent` because I couldn't image a case where you need to go multiple async levels to find the first initiator...but, in theory it's possible, and that code is [already in `PageDependencyGraph`](https://github.com/GoogleChrome/lighthouse/blob/2f6ed6fb2c586f2d8adbdcdcadbad28aad6fb129/core/computed/page-dependency-graph.js#L39-L62), so why not. Note that `getNetworkInitiators` returns an array of initiators because it's interested in _all_ dependencies, but it's a simple iteration through an object so is barely any wasted work.